src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -166,18 +173,36 @@ public Task StartAsync(ITransport transport, HttpClient httpClient)
         {
             try
             {
-                _logger.LogTrace("Beginning receive loop");
+                _logger.LogTrace("Beginning receive loop.");
 
                 while (await Input.WaitToReadAsync())
                 {
+                    if (_connectionState != ConnectionState.Connected)
```

Comment:
I'm pretty sure you can still receive data after the state has shifted into disconnected. Even though the connection state is volatile it doesn't mean that right after this check the state can't shift into disconnected.